.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,56 @@
+name: Build and Test
+
+on: pull_request  
+
+permissions:
+  contents: read
+
+jobs:
+  mac_build_and_test:
+    if: github.repository == 'ml-explore/mlx-swift-examples'
+    runs-on: [self-hosted, macos]
+    steps:
+      - uses: actions/checkout@v6
+        with:
+          submodules: recursive
+     
+      - name: Verify MetalToolchain installed
+        shell: bash
+        run: xcodebuild -showComponent MetalToolchain
+      
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v6
+        with:
+            python-version: ${{ inputs.python-version }}
```

Comment:
This input is never defined. Do we use this whole setup just for pre-commit? Would it make sense to split the workflow into a `lint` job that runs precommit (maybe on ubuntu-latest?) and then unblocks a full macOS build?